src/components/layout/header.js

```diff
@@ -82,35 +82,52 @@ const HeaderSearch = ({ children }) => {
   )
 }
 
-const MobileMenu = () => {
-  const searchInputRef = useRef()
-  const [, searchDispatch] = useSearch()
+const MobileMenu = ({ expanded }) => {
+  const [searchState] = useSearch()
+  const { query, isFetching } = searchState
+  const [menuHeight, setMenuHeight] = useState({ initial: 0, current: 0 })
+  const resultPopoverRef = React.createRef()
+  const menuRef = useRef()
+
+  // Set initial menu height value to reset later.
+  useEffect(() => {
+    if (expanded) {
+      setMenuHeight({ ...menuHeight, initial: menuRef.current.offsetHeight })
+    }
+  }, [expanded])
+
+  // When query changes,
+  // either update menu min height (if needed) or reset to initial value
+  useEffect(() => {
+    if (
+      query &&
+      !isFetching &&
+      resultPopoverRef.current &&
+      resultPopoverRef.current.offsetHeight
+    ) {
+      setMenuHeight({
+        ...menuHeight,
+        current: Math.max(
+          resultPopoverRef.current.offsetHeight + 75,
+          menuHeight.initial,
+        ),
+      })
+    } else if (!query) {
+      setMenuHeight({ ...menuHeight, current: menuHeight.initial })
+    }
+  }, [query, isFetching])
 
   return (
-    <div className={headerStyle.mobileMenu}>
+    <div
+      ref={menuRef}
+      className={headerStyle.mobileMenu}
+      style={{
```

Comment:
Now that we're applying show/hide of the menu using styles directly, instead of classes, it means if you tap the search box then resize, the mobile menu gets stuck open. Could we fix this by either applying an event listener to `onresize`, or even better instead of doing it via style, removing the `display` part of the style definition, and add/remove classes like:

className=`${headerStyle.mobileMenu} ${expanded === true && headerStyle.mobileMenuExpanded}`

![image](https://user-images.githubusercontent.com/270536/81496983-881d0700-9289-11ea-9335-33f6f62b6466.png)

Reply:
Good call. Will be fixed tomorrow!

Reply:
Actually had to do both:
* Add proper styling to avoid `style` taking precedence  
* Add a (throttled every 250ms) `resize` listener to actually "un-expand" the menu, so when you're in small res, open the menu, expand the page, and get back to small res, the menu isn't `expanded` anymore.